### PR TITLE
Ensure ordinals are created during release process

### LINKED ISSUE #15806: Ordinals not created
Now that we are in beta the ordinals should have been populated in `util\libcrypto.num` and `util\libssl.num` as part of the `make update` in the release script. For some reason that didn't happen.

### PR DESCRIPTION
**Pull request by mattcaswell**
We introduce a new makefile target "make release-update" that forces
ordinal file renumbering, and also does the fips checksum updates. We
then call that  from the release script.

Fixes #15806

### CODE REVIEW

File: Configurations/unix-Makefile.tmpl [hunk 1]
```diff
@@ -1109,6 +1109,9 @@ uninstall_image_docs:
 # is sensitive to build.info changes.
 update: generate errors ordinals generate_buildinfo
 
+release-update: generate errors renumber generate_buildinfo \
+                update-fips-checksums
```
Comment (levitte) [inline Configurations/unix-Makefile.tmpl:1113]: Hmmmm...

File: dev/release.sh [hunk 2]
```diff
@@ -321,9 +321,7 @@ echo "== Configuring OpenSSL for update and release.  This may take a bit of tim
 
 $VERBOSE "== Checking source file updates and fips checksums"
 
-make update >&42
-
-make update-fips-checksums >&42
+make release-update >&42
```
Comment (levitte) [inline dev/release.sh:324]: Hmmm.....

**Review by levitte — CHANGES_REQUESTED** [review]
This means that we will do the renumber *every* release, even every alpha.  That's a bit brutal, isn't it?  After all, there was a reason for 17213b2ad03fe577c03a3e77142242adde41fa9f, as the sudden numbers caused a number of merge conflicts that were more disruptive than beneficial...

I would be better to do the renumbering under specific conditions.  `dev/release-aux/release-version-fn.sh` serves a few variables, and I would expect that checking if it's the empty string or `"beta"` would be good enough criteria to launch a renumber *separately*...

**Comment by mattcaswell** [comment]
I added a check to avoid doing the renumber in an alpha release. Please take another look.

**Review by paulidale — APPROVED** [review]

**Comment by paulidale** [comment]
Merged.

**Pull request closed without merging**